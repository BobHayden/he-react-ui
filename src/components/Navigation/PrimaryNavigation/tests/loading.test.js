import createTestContext from 'react-cosmos-test/enzyme'
import fixture from '../fixtures/Loading.fixture'

const { mount, getWrapper } = createTestContext({ fixture })

beforeEach(mount)

test('<PrimaryNavigation /> rendered correctly with fixture', () => {
  let wrapper = getWrapper()

  expect(wrapper).toMatchSnapshot()
})
