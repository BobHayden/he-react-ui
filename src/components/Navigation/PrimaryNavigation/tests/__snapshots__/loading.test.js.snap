// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrimaryNavigation /> rendered correctly with fixture 1`] = `
<OnClickOutside(PrimaryNavigation)
  eventTypes={
    Array [
      "mousedown",
      "touchstart",
    ]
  }
  excludeScrollbar={false}
  items={Array []}
  loading={true}
  locations={Array []}
  logoutRoute="/logout"
  outsideClickIgnoreClass="ignore-react-onclickoutside"
  preventDefault={false}
  stopPropagation={false}
>
  <PrimaryNavigation
    bottomKeys={Array []}
    disableOnClickOutside={[Function]}
    enableOnClickOutside={[Function]}
    eventTypes={
      Array [
        "mousedown",
        "touchstart",
      ]
    }
    items={Array []}
    loading={true}
    locations={Array []}
    logo={
      Object {
        "icon": <Icon
          className="logo"
          name="HealthEngine"
        />,
        "route": "/",
      }
    }
    logoutRoute="/logout"
    outsideClickIgnoreClass="ignore-react-onclickoutside"
    preventDefault={false}
    stopPropagation={false}
  >
    <div
      className="outer"
    >
      <div
        className="nav"
      >
        <div
          className="buckets"
        >
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="logoBucket"
            onClick={[Function]}
            to="/"
          >
            <Route
              path="/"
            >
              <Link
                aria-current="true"
                className="logoBucket active"
                onClick={[Function]}
                replace={false}
                style={Object {}}
                to="/"
              >
                <a
                  aria-current="true"
                  className="logoBucket active"
                  href="/"
                  onClick={[Function]}
                  style={Object {}}
                >
                  <div
                    className="logo"
                  >
                    <Icon
                      className="logo"
                      name="HealthEngine"
                    >
                      <HealthEngine
                        className="icon logo"
                      >
                        <svg
                          className="icon logo"
                          version="1.1"
                          viewBox="0 0 48 38"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M12.9079345,32.7452837 C13.1763914,35.6739197 15.6350174,37.9154191 18.5788836,37.9154191 C21.5227499,37.9154191 23.9813759,35.6739197 24.2498328,32.7452837 L24.2498328,25.1600585 L12.9079345,25.1600585 L12.9079345,32.7452837 L12.9079345,32.7452837 Z"
                            fill="#5BC0C9"
                            id="Shape"
                          />
                          <path
                            d="M24.2498328,6.2159583 L24.2498328,6.2159583 C24.4467061,4.06823902 23.4109333,1.99397789 21.575,0.8592795 C19.7390667,-0.275418886 17.4187006,-0.275418886 15.5827673,0.8592795 C13.7468339,1.99397789 12.7110612,4.06823902 12.9079345,6.2159583 L12.9079345,6.2159583 L12.9079345,13.8011836 L24.2498328,13.8011836 L24.2498328,6.2159583 Z"
                            fill="#5BC0C9"
                            id="Shape"
                          />
                          <path
                            d="M37.1177539,25.1410954 L31.4230769,25.1410954 L31.4230769,32.7263207 C31.6915338,35.6549566 34.1501598,37.896456 37.0940261,37.896456 C40.0378923,37.896456 42.4965183,35.6549566 42.7649752,32.7263207 L42.7649752,19.4521764 C42.7650838,22.5756585 40.2442888,25.1150664 37.1177539,25.1410954 L37.1177539,25.1410954 Z"
                            fill="#C6E400"
                            id="Shape"
                          />
                          <path
                            d="M37.1177539,0.546002402 C33.9726707,0.546002402 31.4230769,3.09301818 31.4230769,6.23492137 L31.4230769,13.8201467 L37.1177539,13.8201467 C40.2628372,13.8201467 42.8124309,16.3671624 42.8124309,19.5090656 L42.8124309,6.23492137 C42.8124309,3.09301818 40.2628372,0.546002402 37.1177539,0.546002402 L37.1177539,0.546002402 Z"
                            fill="#C6E400"
                            id="Shape"
                          />
                          <path
                            d="M5.29604959,13.8201467 C2.36444943,14.0883321 0.120681325,16.5444721 0.120681325,19.4853618 C0.120681325,22.4262514 2.36444943,24.8823915 5.29604959,25.1505769 L12.8889522,25.1505769 L12.8889522,13.8201467 L5.29604959,13.8201467 L5.29604959,13.8201467 Z"
                            fill="#2B4B5A"
                            id="Shape"
                          />
                          <rect
                            fill="#2B4B5A"
                            height="11.3304303"
                            id="Rectangle-path"
                            width="7.6118849"
                            x="24.2498328"
                            y="13.8201467"
                          />
                          <rect
                            fill="#008EC8"
                            height="11.3304303"
                            id="Rectangle-path"
                            width="11.3418983"
                            x="12.9079345"
                            y="13.8201467"
                          />
                          <path
                            d="M37.5279213,13.8201467 L31.8332443,13.8201467 L31.8332443,25.1410954 L37.5279213,25.1410954 C40.4381324,24.8492731 42.6538429,22.4024989 42.6538429,19.480621 C42.6538429,16.5587431 40.4381324,14.111969 37.5279213,13.8201467 L37.5279213,13.8201467 Z"
                            fill="#48B400"
                            id="Shape"
                          />
                        </svg>
                      </HealthEngine>
                    </Icon>
                  </div>
                </a>
              </Link>
            </Route>
          </NavLink>
          <LoadingStrip
            className="loadingBucket"
          >
            <div
              className="outer loadingBucket"
            >
              <div
                className="inner"
              />
            </div>
          </LoadingStrip>
          <LoadingStrip
            className="loadingBucket"
          >
            <div
              className="outer loadingBucket"
            >
              <div
                className="inner"
              />
            </div>
          </LoadingStrip>
          <LoadingStrip
            className="loadingBucket bottom"
          >
            <div
              className="outer loadingBucket bottom"
            >
              <div
                className="inner"
              />
            </div>
          </LoadingStrip>
        </div>
      </div>
      <div
        className="spacer"
      >
        Â 
      </div>
      <div
        className="content"
        onClick={[Function]}
      >
        <withRouter(SubNavigation)
          loading={true}
          logoutRoute="/logout"
        >
          <Route
            render={[Function]}
          >
            <SubNavigation
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "canGo": [Function],
                  "createHref": [Function],
                  "entries": Array [
                    Object {
                      "hash": "",
                      "key": "mocked",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                  ],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "index": 0,
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "key": "mocked",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              loading={true}
              location={
                Object {
                  "hash": "",
                  "key": "mocked",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              logoutRoute="/logout"
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
            >
              <div
                className="bar"
              >
                <div
                  className="top"
                >
                  <Heading
                    className="heading"
                    h1={true}
                    h2={false}
                    h3={false}
                    h4={false}
                    nav={false}
                  >
                    <h1
                      className="heading h1 heading"
                    >
                      <LoadingStrip
                        className="loading"
                      >
                        <div
                          className="outer loading"
                        >
                          <div
                            className="inner"
                          />
                        </div>
                      </LoadingStrip>
                    </h1>
                  </Heading>
                  <div
                    className="controls"
                  >
                    <NavLink
                      activeClassName="active"
                      ariaCurrent="true"
                      className="topNavLink"
                      key="logout"
                      title="Logout"
                      to="/logout"
                    >
                      <Route
                        path="/logout"
                      >
                        <Link
                          aria-current={false}
                          className="topNavLink"
                          replace={false}
                          title="Logout"
                          to="/logout"
                        >
                          <a
                            aria-current={false}
                            className="topNavLink"
                            href="/logout"
                            onClick={[Function]}
                            title="Logout"
                          >
                            <span
                              className="rightControlOption"
                            >
                              <Icon
                                className="icon"
                                name="Logout"
                              >
                                <Logout
                                  className="icon icon"
                                >
                                  <svg
                                    className="icon icon"
                                    version="1.1"
                                    viewBox="0 1 16 16"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <path
                                      d="M 7.5 0.66601562 C 6.6774686 0.66601562 6 1.3434842 6 2.1660156 L 6 5.1660156 L 7 5.1660156 L 7 2.1660156 C 7 1.884547 7.2185314 1.6660156 7.5 1.6660156 L 14.5 1.6660156 C 14.781469 1.6660156 15 1.884547 15 2.1660156 L 15 14.166016 C 15 14.447484 14.781469 14.666016 14.5 14.666016 L 7.5 14.666016 C 7.2185314 14.666016 7 14.447484 7 14.166016 L 7 11.166016 L 6 11.166016 L 6 14.166016 C 6 14.988547 6.6774686 15.666016 7.5 15.666016 L 14.5 15.666016 C 15.322531 15.666016 16 14.988547 16 14.166016 L 16 2.1660156 C 16 1.3434842 15.322531 0.66601562 14.5 0.66601562 L 7.5 0.66601562 z M 3.1464844 4.8125 L -0.20703125 8.1660156 L 3.1464844 11.519531 L 3.8535156 10.8125 L 1.7070312 8.6660156 L 11.5 8.6660156 L 11.5 7.6660156 L 1.7070312 7.6660156 L 3.8535156 5.5195312 L 3.1464844 4.8125 z "
                                      transform="translate(0,1)"
                                    />
                                  </svg>
                                </Logout>
                              </Icon>
                              Logout
                            </span>
                          </a>
                        </Link>
                      </Route>
                    </NavLink>
                  </div>
                </div>
              </div>
            </SubNavigation>
          </Route>
        </withRouter(SubNavigation)>
      </div>
    </div>
  </PrimaryNavigation>
</OnClickOutside(PrimaryNavigation)>
`;
